packages/mobile/src/hooks/useFetchAllFavoritedTrackIds.ts

```diff
@@ -9,38 +9,42 @@ import { useIsOfflineModeEnabled } from './useIsOfflineModeEnabled'
 
 const { getUserId } = accountSelectors
 
+export const fetchAllFavoritedTrackIds = async (currentUserId: number) => {
+  let trackIds: number[] = []
+  let loadMore = true
+  let offset = 0
+  // TODO: store results in state to avoid duplicate fetching
+  while (loadMore) {
+    const url = apiClient.makeUrl(
+      `/users/${encodeHashId(currentUserId)}/favorites`,
+      {
+        user_id: currentUserId,
+        limit: 500,
+        offset
+      }
+    )
+    const { data: result } = await fetch(url).then((response) =>
+      response.json()
+    )
+
+    loadMore = result.length > 0
+    offset += result.length
+    trackIds = trackIds.concat(
+      result
+        .filter((trackSave) => trackSave.favorite_type === 'SaveType.track')
```

Comment:
does this handle the tracks that have been deleted? or guessing that doesn't come back from the api?